packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1575,67 +1576,45 @@ export async function findAndModifyWithFLEFallback(
   object: { $set?: BSONObject; $unset?: BSONObject } | BSONObject | BSONArray,
   modificationType: 'update' | 'replace'
 ): Promise<ErrorOrResult> {
-  let options: { returnDocument: 'before' | 'after'; promoteValues: false } = {
-    returnDocument: 'after',
-    promoteValues: false,
-  };
-  let error: (Error & { codeName?: string; code?: any }) | undefined;
-  let document;
-
   const findOneAndModifyMethod =
     modificationType === 'update' ? 'findOneAndUpdate' : 'findOneAndReplace';
-  const modifyOneMethod =
-    modificationType === 'update' ? 'updateOne' : 'replaceOne';
+  let error: (Error & { codeName?: string; code?: any }) | undefined;
 
   try {
-    document = await ds[findOneAndModifyMethod](ns, query, object, options);
+    return [
+      undefined,
+      await ds[findOneAndModifyMethod](ns, query, object, {
+        returnDocument: 'after',
+        promoteValues: false,
+      }),
+    ] as ErrorOrResult;
   } catch (e) {
     error = e as Error;
   }
 
-  if (!error) {
-    return [undefined, document] as ErrorOrResult;
-  }
+  if (
+    error.codeName === 'ShardKeyNotFound' ||
+    +(error?.code ?? 0) === 63714_02
+  ) {
+    const modifyOneMethod =
+      modificationType === 'update' ? 'updateOne' : 'replaceOne';
 
-  if (error.codeName === 'ShardKeyNotFound') {
     try {
-      const docs = await ds.find(ns, query, options);
-      [document] = docs;
+      await ds[modifyOneMethod](ns, query, object);
     } catch (e) {
       return [e, undefined] as ErrorOrResult;
     }
 
-    if (document) {
-      try {
-        await ds[modifyOneMethod](ns, { _id: document._id }, object);
-      } catch (e) {
-        return [e, undefined] as ErrorOrResult;
-      }
-    }
-  } else if (+(error?.code ?? 0) === 63714_02) {
-    // 6371402 is "'findAndModify with encryption only supports new: false'"
-    options = {
-      returnDocument: 'before',
-      promoteValues: false,
-    };
     try {
-      document = await ds[findOneAndModifyMethod](ns, query, object, options);
-    } catch (e) {
-      // Return the current findOneAndModify error here
-      // since we already know the exact error from the previous attempt
-      // and want to know what went wrong with the second attempt with the fallback options,
-      // e.g. return the `Found indexed encrypted fields but could not find __safeContent__` error.
-      return [e, undefined] as ErrorOrResult;
-    }
-  }
-
-  try {
-    if (document) {
-      const docs = await ds.find(ns, { _id: document._id }, options);
+      const docs = await ds.find(
+        ns,
+        { _id: query._id as ObjectId },
```

Comment:
Did something require this type change? `_id` is not necessarily an `ObjectId` and the types should account for that

Reply:
TS complins then:

Type 'unknown' is not assignable to type 'Condition<ObjectId> | undefined'.ts(2322)
mongodb.d.ts(6272, 5): The expected type comes from property '_id' which is declared here on type 'Filter<Document>'

Reply:
Mh yeah ... I think `as any` would be more appropriate here because this could really be any BSON value. Not blocking though